test/rekt/resources/broker/broker.go

```diff
@@ -18,24 +18,121 @@ package broker
 
 import (
 	"context"
+	"fmt"
 	"testing"
+	"time"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	eventingv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/k8s"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
 	"knative.dev/reconciler-test/pkg/feature"
 	"knative.dev/reconciler-test/pkg/manifest"
 )
 
-// Output is the base output we can expect from a echo job.
-type Output struct {
-	Success bool   `json:"success"`
-	Message string `json:"msg"`
+type CfgFn func(map[string]interface{})
+
+// WithBrokerClass adds the broker class config to a Broker spec.
+func WithBrokerClass(class string) CfgFn {
+	return func(cfg map[string]interface{}) {
+		cfg["brokerClass"] = class
+	}
 }
 
-func Install(name string) feature.StepFn {
+// WithDeadLetterSink adds the dead letter sink related config to a Broker spec.
+func WithDeadLetterSink(ref *duckv1.KReference, uri string) CfgFn {
+	return func(cfg map[string]interface{}) {
+		if _, set := cfg["delivery"]; !set {
+			cfg["delivery"] = map[string]interface{}{}
+		}
+		delivery := cfg["delivery"].(map[string]interface{})
+		if _, set := delivery["deadLetterSink"]; !set {
+			delivery["deadLetterSink"] = map[string]interface{}{}
+		}
+		dls := delivery["deadLetterSink"].(map[string]interface{})
+		if uri != "" {
+			dls["uri"] = uri
+		}
+		if ref != nil {
+			if _, set := dls["ref"]; !set {
+				dls["ref"] = map[string]interface{}{}
+			}
+			dref := dls["ref"].(map[string]interface{})
+			dref["apiVersion"] = ref.APIVersion
+			dref["kind"] = ref.Kind
+			// Skip namespace.
+			dref["name"] = ref.Name
+		}
+	}
+}
+
+// WithRetry adds the retry related config to a Broker spec.
+func WithRetry(count int32, backoffPolicy *eventingv1.BackoffPolicyType, backoffDelay *string) CfgFn {
+	return func(cfg map[string]interface{}) {
+		if _, set := cfg["delivery"]; !set {
+			cfg["delivery"] = map[string]interface{}{}
+		}
+		delivery := cfg["delivery"].(map[string]interface{})
+
+		delivery["retry"] = count
+		if backoffPolicy != nil {
+			delivery["backoffPolicy"] = backoffPolicy
+		}
+		if backoffDelay != nil {
+			delivery["backoffDelay"] = backoffDelay
+		}
+	}
+}
```

Comment:
Maybe not related to this code itself, but in some use cases would it make sense to get, as options, directly the api types? Like in this case, you could get `DeliverySpec` and then somehow we smash it in the json... Can we have something from reconciler-test side that allows us to smash api types inside the template?

Reply:
I wanted to keep the core resources as typed as possible without using too much of the types directly.

Reply:
Then can we put those functions in a reusable package outside `broker`? 
They seem generic enough to be reused for channel tests.

Reply:
This is specific for the broker template in play. I am building an api around interacting with the broker.yaml template